# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.cpp
# Commit: f878938d8cb9
# Full Hash: f878938d8cb96ac802bbd43bdbbc210439946671
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-10-06 09:37:17
# Description:
#   Bug 1793423: Don't assume the document has a cache yet in RemoteAccessibleBase::ChildAtPoint. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D158623
# ==============================================================================

diff -r 73b5d1dba5af -r f878938d8cb9 accessible/ipc/RemoteAccessibleBase.cpp
--- a/accessible/ipc/RemoteAccessibleBase.cpp	Wed Oct 05 23:56:22 2022 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.cpp	Wed Oct 05 23:57:11 2022 +0000
@@ -349,6 +349,11 @@
   // If `this` is a document, use its viewport cache instead of
   // the cache of its parent document.
   if (DocAccessibleParent* doc = IsDoc() ? AsDoc() : mDoc) {
+    if (!doc->mCachedFields) {
+      // A client call might arrive after we've constructed doc but before we
+      // get a cache push for it.
+      return nullptr;
+    }
     if (auto maybeViewportCache =
             doc->mCachedFields->GetAttribute<nsTArray<uint64_t>>(
                 nsGkAtoms::viewport)) {
